packages/synapse/src/Synapse.test.ts

```diff
@@ -0,0 +1,235 @@
+          },
+        })
+      }
+    test(
+      'Should return a valid transaction for a L2 -> L1 transaction (Non CCTP)',
+    ),
+      async () => {
+        const filter = await bridge({
+          sourceChainId: ARBITRUM_CHAIN_ID,
+          destinationChainId: ETH_CHAIN_ID,
+          tokenAddress: ARBITRUM_USDCE_ADDRESS,
+          amount: GreaterThanOrEqual(100000n),
+          recipient: TEST_USER,
+        })
+
+        expect(filter).to.deep.equal({
+          chainId: ARBITRUM_CHAIN_ID,
+          to: CHAIN_TO_ROUTER[ARBITRUM_CHAIN_ID],
+          input: {
+            $abi: SYNAPSE_BRIDGE_FRAGMENTS,
+            to: TEST_USER,
+            amount: {
+              $gte: '100000',
+            },
+            chainId: ETH_CHAIN_ID,
+            token: ARBITRUM_USDCE_ADDRESS,
+          },
+        })
+      }
+    // CCTP Transactions
+    test('Should return a valid transaction for a L1 -> L2 transaction (CCTP)'),
+      async () => {
+        const filter = await bridge({
+          sourceChainId: ETH_CHAIN_ID,
+          destinationChainId: ARBITRUM_CHAIN_ID,
+          tokenAddress: ETHEREUM_USDC_ADDRESS,
+          amount: GreaterThanOrEqual(100000n),
+          recipient: TEST_USER,
+          contractAddress: getContractAddress(ETH_CHAIN_ID),
+        })
+
+        expect(filter).to.deep.equal({
+          chainId: ETH_CHAIN_ID,
+          // Update
+          to: SynapseCCTPContract[ETH_CHAIN_ID],
+          input: {
+            $abi: SYNAPSE_BRIDGE_FRAGMENTS,
+            //Update
+            sender: TEST_USER,
+            amount: {
+              $gte: '100000',
+            },
+            chainId: ARBITRUM_CHAIN_ID,
+            token: ETHEREUM_USDC_ADDRESS,
+          },
+        })
+      }
+    test('Should return a valid transaction for a L2 -> L1 transaction (CCTP)'),
+      async () => {
+        const filter = await bridge({
+          sourceChainId: ARBITRUM_CHAIN_ID,
+          destinationChainId: ETH_CHAIN_ID,
+          tokenAddress: ARBITRUM_USDC_ADDRESS,
+          amount: GreaterThanOrEqual(100000n),
+          recipient: TEST_USER,
+          contractAddress: getContractAddress(ETH_CHAIN_ID),
+        })
+
+        expect(filter).to.deep.equal({
+          chainId: ARBITRUM_CHAIN_ID,
+          to: SynapseCCTPContract[ARBITRUM_CHAIN_ID],
+          input: {
+            $abi: SYNAPSE_BRIDGE_FRAGMENTS,
+            sender: TEST_USER,
+            amount: {
+              $gte: '100000',
+            },
+            chainId: ETH_CHAIN_ID,
+            token: ARBITRUM_USDC_ADDRESS,
+          },
+        })
+      }
+
+    describe('When applying the filter', () => {
+      test('should pass filter with valid L1 ETH tx', async () => {
+        const transaction = DEPOSIT_ETH
+        const filter = await bridge({
+          sourceChainId: ETH_CHAIN_ID,
+          destinationChainId: ARBITRUM_CHAIN_ID,
+          tokenAddress: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
```

Comment:
For ETH to work properly on our front end, it needs to use the zero address.

You will need to convert it back to 0xEeee in the bridge function before returning the filter.